pkg/service/FluxService/FluxApplicationService.go

```diff
@@ -0,0 +1,420 @@
+			if err != nil {
+				impl.logger.Errorw("error in getting cluster repository", "err", err)
+			}
+			clusterInfo := impl.converter.GetClusterInfo(cluster)
+			clusterConfig := impl.converter.GetClusterConfig(clusterInfo)
+			restConfig, err := impl.k8sUtil.GetRestConfigByCluster(clusterConfig)
+			if err != nil {
+				impl.logger.Errorw("error in getting cluster config", "err", err)
+			}
+			resp, err := impl.k8sUtil.GetResource(context.Background(), app.EnvironmentDetails.Namespace, app.Name, GvkForKustomizationFluxApp, restConfig)
+			if err != nil || resp == nil {
+				impl.logger.Errorw("error in getting response", "err", err)
+			}
+
+			inventoryMap, err := getInventoryMap(resp.Manifest.Object)
+			if err != nil {
+				impl.logger.Errorw("error in inventories", "err", err)
+			}
+
+			for id, version := range inventoryMap {
+				var fluxResource FluxKsResourceDetail
+				fluxResource, err = parseObjMetadata(id)
+				if err != nil {
+					fmt.Println("issue is here for some reason , r", err)
+				}
+				fluxResource.Version = version
+				if fluxResource.Group == FluxKustomizationGroup && fluxResource.Kind == FluxAppKustomizationKind {
+					if fluxResource.Name == app.Name && fluxResource.Namespace == app.EnvironmentDetails.Namespace {
+						continue
+					}
+					childParentMap[fluxResource.Name] = true
+				}
+			}
+		}
+	}
+
+	for _, app := range appList {
+		if app.Name == "flux-system" && app.EnvironmentDetails.Namespace == "flux-system" {
+			continue
+		} else if childParentMap[app.Name] != true {
+			childFilteredAppList = append(childFilteredAppList, kustomizationMap[app.Name])
+		}
+	}
+	return childFilteredAppList
+}
+func (impl *FluxApplicationServiceImpl) buildFluxAppDetailForHelmRelease(request *FluxAppDetailRequest) ([]*bean.ResourceTreeResponse, *FluxAppStatusDetail, error) {
+	fluxAppTreeResponse := make([]*bean.ResourceTreeResponse, 0)
+
+	k8sConfig := impl.converter.GetClusterConfigFromClientBean(request.Config)
+	restConfig, err := impl.k8sUtil.GetRestConfigByCluster(k8sConfig)
+	resp, err := impl.k8sUtil.GetResource(context.Background(), request.Namespace, request.Name, GvkForHelmreleaseFluxApp, restConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting resource", "err", err)
+		return nil, nil, err
+	}
+
+	releaseName := ""
+	_namespace := ""
+	var appStatus *FluxAppStatusDetail
+
+	if resp != nil && resp.Manifest.Object != nil {
+		releaseName, _namespace, err = getReleaseNameNamespace(resp.Manifest.Object, request.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		appStatus, err = getKsAppStatus(resp.Manifest.Object, GvkForHelmreleaseFluxApp)
+		if err != nil {
+			impl.logger.Errorw("error in getting app status", "err", err)
+			return nil, nil, err
+		}
+
+		if releaseName == "" && _namespace == "" {
+			impl.logger.Errorw("error in getting release name and namespace", "err", err)
+			return nil, appStatus, err
+		}
```

Comment:
I think this is wrong maybe we can throw custom error in this case if an error is supposed to be returned, since this err is already handled above. also implement the behaviour for appStatus as we discussed